pkg/util/informermanager/informermanager_test.go

```diff
@@ -0,0 +1,950 @@
+	}).WithTimeout(time.Second * 2).Should(gomega.Succeed())
+}
+
+// Verifies that the event handler from an EventHandlerGenerator is regenerated and registered on a corresponding FTC
+// update where Predicate returns true and Generator returns an event handler.
+func TestInformerManagerEventHandlerReregisteredFTCUpdate(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	// 1. Bootstrap an environemnt with a single Deployment FTC and a single deployment.
+
+	ftc := deploymentFTC.DeepCopy()
+	dp1 := getDeployment("dp-1", "default")
+
+	defaultFTCs := []*fedcorev1a1.FederatedTypeConfig{ftc}
+	defaultObjects := []*unstructured.Unstructured{dp1}
+	manager, dynamicClient, fedClient := boostrapInformerManagerWithFakeClients(defaultFTCs, defaultObjects)
+
+	// 2. Add EventHandlerGenerators to the InformerManager, eventHandler SHOULD always be regenerated and registered.
+
+	eventHandler := &countingResourceEventHandler{}
+	manager.AddEventHandlerGenerator(&EventHandlerGenerator{
+		Predicate: func(lastApplied *fedcorev1a1.FederatedTypeConfig, latest *fedcorev1a1.FederatedTypeConfig) bool {
+			return true
+		},
+		Generator: eventHandler.generateEventHandler,
+	})
+
+	// 3. Start InformerManager.
+
+	ctx := context.Background()
+	manager.Start(ctx)
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, time.Second*3)
+	defer cancel()
+	if !cache.WaitForCacheSync(ctxWithTimeout.Done(), manager.HasSynced) {
+		g.Fail("Timed out waiting for InformerManager cache sync")
+	}
+
+	// 4. Verify that eventHandler is generated and registered initially.
+
+	g.Eventually(func(g gomega.Gomega) {
+		g.Expect(eventHandler.getGenerateCount()).To(gomega.BeNumerically("==", 1))
+		g.Expect(eventHandler.getAddEventCount()).To(gomega.BeNumerically("==", 1))
+		g.Expect(eventHandler.getUpdateEventCount()).To(gomega.BeZero())
+		g.Expect(eventHandler.getDeleteEventCount()).To(gomega.BeZero())
+	}).WithTimeout(time.Second * 2).Should(gomega.Succeed())
+
+	// 5. Trigger FTC updates
+
+	for i := 0; i < 5; i++ {
+		ftc.SetAnnotations(map[string]string{"trigger": strconv.Itoa(rand.Intn(1000))})
+		_, err := fedClient.CoreV1alpha1().FederatedTypeConfigs().Update(ctx, ftc, metav1.UpdateOptions{})
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+
+		// 5a. Generate deployment events as well
+
+		// +1 Update event
+
+		dp1.SetAnnotations(map[string]string{"trigger": strconv.Itoa(rand.Intn(1000))})
+		dp1, err = dynamicClient.Resource(common.DeploymentGVR).Namespace("default").Update(ctx, dp1, metav1.UpdateOptions{})
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+
+		// 5b. Verify eventHandler is regenerated and registered
+
+		g.Eventually(func(g gomega.Gomega) {
+			g.Expect(eventHandler.getGenerateCount()).To(gomega.BeNumerically("==", 1+i))
+			g.Expect(eventHandler.getAddEventCount()).To(gomega.BeNumerically("==", 1))
+			g.Expect(eventHandler.getUpdateEventCount()).To(gomega.BeNumerically("==", i))
+			g.Expect(eventHandler.getDeleteEventCount()).To(gomega.BeZero())
+		})
```

Comment:
Missed success check

Reply:
Pls do a sweep

Reply:
i will resolve this issue first, since it is captured by the linter now